### PR TITLE
Enhancements to `SparseWeightsBase`

### PR DESCRIPTION
**Pull request by mvacaporale**
Changes include:
* Allow edge case sparsity (`assert 0 <= sparsity <= 1`)
* Introduce `off_mask` setter and getter attributes
* Tests for the added/modified functionality

### CODE REVIEW

File: nupic/torch/modules/sparse_weights.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def __init__(self, module, weight_sparsity=None, sparsity=None):
                 DeprecationWarning,
             )
 
-        assert 0 < sparsity < 1
+        assert 0 <= sparsity <= 1
```
Comment (lscheinkman) [inline nupic/torch/modules/sparse_weights.py:83]: Why do you need these edge cases? 

`sparsity=0` means dense
`sparsity=1` means all zeros

Not sure if any of these cases make sense in the context of the `SparseWeights`

File: tests/sparse_weights_test.py [hunk 2]
```diff
@@ -100,6 +100,78 @@ def test_rezero_after_forward_2d(self):
                 expected = [round(input_size * (1.0 - sparsity))] * out_channels
                 self.assertSequenceEqual(counts.numpy().tolist(), expected)
 
+    def test_fully_dense_linear(self):
+
+        lin = torch.nn.Linear(10, 10)
+        sw = SparseWeights(lin, sparsity=0)
```
Comment (lscheinkman) [inline tests/sparse_weights_test.py:106]: Wrapping a dense layer with `sparsity=0` makes no sense. It wont change anything on the original layer

File: tests/sparse_weights_test.py [hunk 3]
```diff
@@ -100,6 +100,78 @@ def test_rezero_after_forward_2d(self):
...
+
+        self.assertTrue(torch.all(sw.off_mask == torch.zeros_like(sw.weight)))
+        sw.weight[:] = 1
+        sw.rezero_weights()
+        self.assertTrue(sw.weight.sum() == sw.weight.numel())
+
+    def test_fully_sparse_linear(self):
+
+        lin = torch.nn.Linear(10, 10)
+        sw = SparseWeights(lin, sparsity=1)
```
Comment (lscheinkman) [inline tests/sparse_weights_test.py:116]: Wrapping a dense layer with sparsity=1 makes no sense. It is the same as removing the original layer

File: nupic/torch/modules/sparse_weights.py [hunk 4]
```diff
@@ -119,6 +119,25 @@ def rezero_weights(self):
         """
         raise NotImplementedError
 
+    @property
+    def weight(self):
+        return self.module.weight
+
+    @property
+    def off_mask(self):
```
Comment (lscheinkman) [inline nupic/torch/modules/sparse_weights.py:127]: Could you add a docsting describing this property?
